### PR TITLE
Python 3 11 support

### PR DESCRIPTION
**Pull request by elijahbenizzy**
[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
Removed tests, decided not to make it impossible to install as I don't want to create a new major version. Next major version release will probably be. 3.8 or 3.10 +.

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
       task:
         type: string
     docker:
-      - image: circleci/python:<< parameters.python-version >>
+      - image: cimg/python:<< parameters.python-version >>
```
Comment (skrawcz) [inline .circleci/config.yml:10]: ?

**Review by skrawcz — CHANGES_REQUESTED** [review]
I don't see how this fulfills requirements for 3.11 and removing 3.6

1. can the 3.6 commit be separate please?
2. I think there's a few changes that are missing:
* setup.py is not modified.
* requirements.txt isn't updated
* references to 3.6 python aren't removed -- line 79 in hamilton/base.py, line 91 in test_h_async.py

**Comment by elijahbenizzy** [comment]
> I don't see how this fulfills requirements for 3.11 and removing 3.6
> 
> 1. can the 3.6 commit be separate please?
> 2. I think there's a few changes that are missing:
> 
> * setup.py is not modified.
> * requirements.txt isn't updated
> * references to 3.6 python aren't removed -- line 79 in hamilton/base.py, line 91 in test_h_async.py

Can split out commits. See https://github.com/stitchfix/hamilton/pull/291#issuecomment-1407760328 for why I didn't *fully* remove 3.6 support. I think we'd have to cut a major version or break the semantic versioning rules. I guess we could cut a minor one...

File: .circleci/config.yml (hunk 1 above)
Comment (elijahbenizzy) [reply .circleci/config.yml:10]: This worked, the other one didn't :)

**Comment by elijahbenizzy** [comment]
Think I got everything

**Comment by skrawcz** [comment]
> > I don't see how this fulfills requirements for 3.11 and removing 3.6
> > 
> > 1. can the 3.6 commit be separate please?
> > 2. I think there's a few changes that are missing:
> > 
> > 
> > * setup.py is not modified.
> > * requirements.txt isn't updated
> > * references to 3.6 python aren't removed -- line 79 in hamilton/base.py, line 91 in test_h_async.py
> 
> Can split out commits. See [#291 (comment)](https://github.com/stitchfix/hamilton/pull/291#issuecomment-1407760328) for why I didn't _fully_ remove 3.6 support. I think we'd have to cut a major version or break the semantic versioning rules. I guess we could cut a minor one...

As discussed. I think we're fine. Python 3.6 people just wont get new updates -- so no broken functionality for them.

**Comment by skrawcz** [comment]
@elijahbenizzy just requirements.txt -- doesn't need dataclasses conditional install anymore

**Review by skrawcz — APPROVED** [review]

**Pull request merged**